routers/web/repo/recent_commits.go

```diff
@@ -0,0 +1,41 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package repo
+
+import (
+	"errors"
+	"net/http"
+
+	"code.gitea.io/gitea/modules/base"
+	"code.gitea.io/gitea/modules/context"
+	contributors_service "code.gitea.io/gitea/services/repository"
+)
+
+const (
+	tplRecentCommits base.TplName = "repo/activity"
+)
+
+// RecentCommits renders the page to show recent commit frequency on repository
+func RecentCommits(ctx *context.Context) {
+	ctx.Data["Title"] = ctx.Tr("repo.activity.navbar.recent_commits")
+
+	ctx.Data["PageIsActivity"] = true
+	ctx.Data["PageIsRecentCommits"] = true
+	ctx.PageData["repoLink"] = ctx.Repo.RepoLink
```

Comment:
```suggestion
```

I would suggest avoid using PageData here, just use `data-repo-link` to pass the data for the component, it makes it easier to maintain.

Reply:
Or, at least, it should use a "namesapce" like the guideline says:  `ctx.PageData["myModuleData"] = map[]{...}`. https://docs.gitea.com/contributing/guidelines-frontend

Otherwise, if there are more than one vue component in one page, putting all their data in PageData would cause conflicts.

Reply:
> just use `data-repo-link` to pass the data for the component

How do I pass a dynamic value (repo link) inside a `tmpl` file? The locale strings are set inside `/templates/repo/recent_commits.tmpl`.

Reply:
I think you have used it somewhere 😁  this is only a nit, doesn't block the LGTM/merge. I could propose some more fixes in following PRs.

![image](https://github.com/go-gitea/gitea/assets/2114189/d9ee7f98-04f4-4b59-afbe-7a9154013ef7)

Reply:
Oh, wait. I can access `ctx` :D

Reply:
I've done [this](https://github.com/go-gitea/gitea/pull/29210/commits/f8771d5a9ee189bd6754f04e89329c2848a69101) but it says "`repoLink` is undefined". ![image](https://github.com/go-gitea/gitea/assets/32161460/5428fb8b-f7ab-4ee9-bd59-73d2f230c05b)

It seems that the value of repoLink is not set when I try to fetch data with `response = await GET(`${repoLink}/activity/recent-commits/data`)`. Any suggestions?

Reply:
@wxiaoguang the issue is not fixed. Please see my comment above.

Reply:
No worry, you could use the old approach, I will take a look later.